src/htmerl_sax_utf8.erl

```diff
@@ -70,6 +70,14 @@ string(Bin, Options) ->
    Bin1 = norm_newlines(Bin, <<>>),
    data(Bin1, M3).
 
+norm_whitespaces(Bin) ->
+   Splitted = binary:split(Bin, [<<"\n">>, <<" ">>, <<"\t">>], [global, trim_all]),
+   list_to_binary(combine(Splitted, " ")).
+
+combine([], _) -> "" ;
+combine([Head|Tail], Space) -> 
+   binary_to_list(Head) ++ Space ++ combine(Tail, Space).
```

Comment:
To avoid turning binaries into lists, appending them, and then back to binary...
Here an IO List can be built once and flattened once. It will be a bit nicer on memory and speed.

```erlang
  iolist_to_binary(combine(Splitted, <<" ">>)).

combine([], _) -> [];
combine([Last], _) -> Last;
combine([Head|Tail], Space) ->
   [Head, Space, combine(Tail, Space)].
```